### PR TITLE
feat: [AXIMST-617] display section and sequence sidebar level

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** display section and sequence sidebar level

**Youtrack:** https://youtrack.raccoongang.com/issue/AXIMST-617

### CODE REVIEW

File: src/course-home/outline-tab/SequenceLink.jsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useContext } from 'react';
+import React from 'react';
```
Comment (PKulkoRaccoonGang) [inline src/course-home/outline-tab/SequenceLink.jsx:1]: [question]: Do we need this import?

File: src/course-home/outline-tab/SequenceLink.jsx [hunk 2]
```diff
@@ -94,44 +84,42 @@ const SequenceLink = ({
...
+    <li>
+      <div className={classNames('', { 'mt-2 pt-2 border-top border-light': !first })}>
+        <div className="row w-100 m-0">
+          <div className="col-auto p-0">
+            {complete ? (
+              <FontAwesomeIcon
+                icon={fasCheckCircle}
+                fixedWidth
+                className="float-left text-success mt-1"
+                aria-hidden="true"
```
Comment (PKulkoRaccoonGang) [inline src/course-home/outline-tab/SequenceLink.jsx:96]: Maybe
```suggestion
                aria-hidden={complete}
```

File: src/course-home/outline-tab/SequenceLink.jsx [hunk 3]
```diff
@@ -94,44 +84,42 @@ const SequenceLink = ({
...
+                title={intl.formatMessage(messages.completedAssignment)}
+              />
+            ) : (
+              <FontAwesomeIcon
+                icon={farCheckCircle}
+                fixedWidth
+                className="float-left text-gray-400 mt-1"
+                aria-hidden="true"
```
Comment (PKulkoRaccoonGang) [inline src/course-home/outline-tab/SequenceLink.jsx:104]:
```suggestion
                aria-hidden={complete}
```

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
+    width: 32.125rem;
+    max-width: 100%;
+    overflow: auto;
+}
+
+.outline-sidebar-heading-wrapper {
+    border: 1px solid #d7d3d1;
+
+    .outline-sidebar-heading {
+        font-weight: 700;
```
Comment (PKulkoRaccoonGang) [inline src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss:11]:
```suggestion
        font-weight: $font-weight-bold;
```

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss [hunk 5]
```diff
@@ -1,19 +1,51 @@
...
 
     li {
         font-size: 1rem;
-        line-height: 1.25rem;
+        line-height: 1.5rem;
 
         .collapsible-trigger {
-            padding: .625rem $spacer;
-            position: relative;
+            padding: 1.375rem 1.5rem 1.375rem 3.375rem;
```
Comment (PKulkoRaccoonGang) [inline src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss:45]: [nit]: Please, try to use `$spacers` from Paragon for all paddings and margins

Screenshot 2024-03-12 at 16 30 18

File: src/courseware/course/sidebar/sidebars/course-outline/Section.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+const Section = ({ intl, section, handleSelectSection }) => {
+  const {
+    id,
+    complete,
+    title,
+  } = section;
+
+  const sectionTitle = (
+    <>
+      <div className="col-auto p-0" style={{ fontSize: '1.1rem' }}>
```
Comment (PKulkoRaccoonGang) [inline src/courseware/course/sidebar/sidebars/course-outline/Section.jsx:21]: [question]: Can we move `style={{ fontSize: '1.1rem' }}` to a styles file?

File: src/courseware/course/sidebar/sidebars/course-outline/Unit.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        return complete ? <EditSquareCompleteIcon /> : <EditSquareIcon />;
+      default:
+        return complete ? <BookCompleteIcon /> : <BookIcon />;
+    }
+  };
+
+  return (
+    <li className={classNames({ 'bg-info-100': isActive, 'border-top border-light': !first })}>
+      <div className="row w-100 m-0 d-flex align-items-center">
+        <div className="col-auto p-0" style={{ fontSize: '18px' }}>
```
Comment (PKulkoRaccoonGang) [inline src/courseware/course/sidebar/sidebars/course-outline/Unit.jsx:45]: [question]: Can we move `style={{ fontSize: '18px' }}` to a styles file?

File: src/courseware/course/sidebar/sidebars/course-outline/icons/BookCompleteIcon.jsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
+const BookCompleteIcon = (props) => (
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/courseware/course/sidebar/sidebars/course-outline/icons/BookCompleteIcon.jsx:1]: This applies to all new icons we add.

**Review by PKulkoRaccoonGang — CHANGES_REQUESTED** [review]

File: src/course-home/outline-tab/SequenceLink.jsx (hunk 1 above)
Comment (ihor-romaniuk) [reply src/course-home/outline-tab/SequenceLink.jsx:1]: Removed

File: src/course-home/outline-tab/SequenceLink.jsx (hunk 2 above)
Comment (ihor-romaniuk) [reply src/course-home/outline-tab/SequenceLink.jsx:96]: Changed

File: src/course-home/outline-tab/SequenceLink.jsx (hunk 3 above)
Comment (ihor-romaniuk) [reply src/course-home/outline-tab/SequenceLink.jsx:104]: Changed

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss (hunk 4 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss:11]: Changed

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss (hunk 5 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.scss:45]: Changed

File: src/courseware/course/sidebar/sidebars/course-outline/Section.jsx (hunk 6 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/Section.jsx:21]: Removed, no longer used

File: src/courseware/course/sidebar/sidebars/course-outline/Sequence.jsx [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  );
+
+  return (
+    <li>
+      <Collapsible
+        className={classNames('mb-2', { 'active-section': id === activeSequenceId })}
+        styling="card-lg"
+        title={sectionTitle}
+        open={open}
+        onToggle={() => { setOpen(!open); }}
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/Sequence.jsx:83]: Simplified (also was done in next PR))

File: src/courseware/course/sidebar/sidebars/course-outline/Unit.jsx (hunk 7 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/Unit.jsx:45]: Removed, no longer used

File: src/courseware/course/sidebar/sidebars/course-outline/icons/BookCompleteIcon.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/icons/BookCompleteIcon.jsx:1]: These icons will be replaced with existing paragon icons.
We are waiting for the design and we will replace them.

**Review by PKulkoRaccoonGang — APPROVED** [review]

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTrigger.jsx [hunk 10]
```diff
@@ -1,32 +1,56 @@
...
+  } = useContext(SidebarContext);
+
+  const isDisplayForDesktopView = !isMobileView && !shouldDisplayFullScreen && currentSidebar !== ID;
+  const isDisplayForMobileView = isMobileView && shouldDisplayFullScreen;
+
+  if (!isDisplayForDesktopView && !isDisplayForMobileView) {
+    return null;
+  }
+
+  const handleToggleCollapse = () => (currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID));
```
Comment (monteri) [inline src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTrigger.jsx:29]:
```suggestion
  const handleToggleCollapse = () => currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID);
```

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.jsx [hunk 11]
```diff
@@ -1,105 +1,134 @@
...
   const {
     sectionId: activeSectionId,
   } = useModel('sequences', activeSequenceId);
 
   const rootCourseId = courseBlocks?.courses && Object.keys(courseBlocks?.courses)[0];
+  const sectionsIds = courseBlocks?.courses[rootCourseId]?.sectionIds;
+  const sequenceIds = courseBlocks?.sections[selectedSection || activeSectionId]?.sequenceIds;
+  const backButtonTitle = courseBlocks?.sections[selectedSection || activeSectionId]?.title;
+
+  const handleToggleCollapse = () => (currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID));
```
Comment (monteri) [inline src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.jsx:49]:
```suggestion
  const handleToggleCollapse = () => currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID);
```

File: src/courseware/course/Course.test.jsx [hunk 12]
```diff
@@ -121,27 +121,32 @@ describe('Course', () => {
 
     const notificationTrigger = screen.getByRole('button', { name: /Show notification tray/i });
     expect(notificationTrigger).toBeInTheDocument();
-    expect(notificationTrigger.parentNode).not.toHaveClass('mt-3', { exact: true });
+    expect(notificationTrigger.parentNode).not.toHaveClass('ml-1 border-primary-700', { exact: true });
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/course/Course.test.jsx:124]: Changed

File: src/courseware/course/sidebar/sidebars/course-outline/Unit.jsx [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+import {
+  EditSquareIcon, EditSquareCompleteIcon, VideocamIcon, VideocamCompleteIcon,
+  BookIcon, BookCompleteIcon,
+} from './icons';
+import messages from './messages';
+
+const MOCKED_TYPE = 'problem'; // ToDo: should be refactored after API is ready
+
+const Unit = ({
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/Unit.jsx:14]: Renamed

File: src/courseware/course/sidebar/sidebars/course-outline/Sequence.jsx [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+  const {
+    courseBlocks: { units = MOCKED_UNITS },
+  } = useModel('outline', courseId);
+
+  const [open, setOpen] = useState(defaultOpen);
+
+  useEffect(() => {
+    setOpen(defaultOpen);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/Sequence.jsx:56]: I tried removing this `useEffect` and the functionality continues to work correctly.

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTrigger.jsx (hunk 10 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTrigger.jsx:29]: We will catch an eslint error `ESLint: Arrow function used ambiguously with a conditional expression.(no-confusing-arrow)`

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.jsx (hunk 11 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.jsx:49]: The same here

**Review by monteri — APPROVED** [review]

**Pull request merged**